### PR TITLE
Fix for debug build: Fix ill-formed KOKKOS macro function decoration.

### PR DESCRIPTION
**Pull request by overfelt**

### CODE REVIEW

File: include/master_element/MasterElementFunctions.h [hunk 1]
```diff
@@ -82,11 +82,11 @@ generic_grad_op(
...
   static_assert(OutputViewType::rank == 3, "Weight view assumed to be rank 3");
 
-  STK_ThrowAssert(
+  STK_NGP_ThrowAssert(
     AlgTraits::nodesPerElement_ == referenceGradWeights.extent(1));
-  STK_ThrowAssert(AlgTraits::nDim_ == referenceGradWeights.extent(2));
+  STK_NGP_ThrowAssert(AlgTraits::nDim_ == referenceGradWeights.extent(2));
   for (int i = 0; i < dim; ++i)
-    STK_ThrowAssert(weights.extent(i) == referenceGradWeights.extent(i));
+    STK_NGP_ThrowAssert(weights.extent(i) == referenceGradWeights.extent(i));
```
Comment (jrood-nrel) [inline include/master_element/MasterElementFunctions.h:89]: I think the NGP string was dropped in the Trilinos we were using (15.1.1 I think)?

Comment (overfelt) [reply include/master_element/MasterElementFunctions.h:89]: Plain STK_ThrowAssert failed in my Cuda debug build. What are we using these days.

Comment (overfelt) [reply include/master_element/MasterElementFunctions.h:89]: I do have Trilinos 15.1.1.

Comment (jrood-nrel) [reply include/master_element/MasterElementFunctions.h:89]: Oh you're right. https://github.com/Exawind/nalu-wind/pull/1255/files

**Review by jrood-nrel — APPROVED** [review]

**Comment by marchdf** [comment]
Closes #1302

**Comment by jrood-nrel** [comment]
This is great. Thanks James.

**Comment by marchdf** [comment]
@overfelt thank you for taking care of this! I would not have gotten there. My GPU dev box is down this week so hopefully I can test it next week.

**Review by marchdf — APPROVED** [review]

**Comment by overfelt** [comment]
It is crazy that on a debug build the function became a no-op. No warning, no error. It just did nothing. Stared at the code for quite a while before it jumped out at me the "KOKKOS_FUNCTION KOKKOS_FUNCTION" in the declaration.

**Comment by marchdf** [comment]
eagle eyes. Thank you very much!

**Pull request merged**